File: python/web_tutorial_mastercopy/array1d.py
import sys
print(sys.path)

from bridges.bridges import *
from bridges.array import *

def main():
    args = sys.argv[1:]

    # create the Bridges object, set credentials

    bridges = Bridges(int(args[0]), args[1], args[2])

    if len(args) > 3:
        bridges.connector.set_server(args[3])

    #Title and Description
    bridges.set_title("One-Dimensional Array Example")
    bridges.set_description("OneD array with colors in order left to right: "
            +	"red, green, blue, cyan, magenta, yellow, red, green, blue, black.")

    #set array dimensions, allocate array fo elements
    arraySize = 10

    arr = Array(num_elements = arraySize)

    #populate the array with squares of indicies
    for i in range(arr.get_size()):
        arr.get_element(i).set_label(i*i)

    #color the array elements
    arr.get_element(0).get_visualizer().set_color("red")
    arr.get_element(1).get_visualizer().set_color("green")
    arr.get_element(2).get_visualizer().set_color("blue")
    arr.get_element(3).get_visualizer().set_color("cyan")
    arr.get_element(4).get_visualizer().set_color("magenta")
    arr.get_element(5).get_visualizer().set_color("yellow")
    arr.get_element(6).get_visualizer().set_color("red")
    arr.get_element(7).get_visualizer().set_color("green")
    arr.get_element(8).get_visualizer().set_color("blue")
    arr.get_element(9).get_visualizer().set_color("black")

    # tell Bridges what data structure to visualize
    bridges.set_data_structure(arr)

    # visualize the list
    bridges.visualize()

if __name__ == "__main__":
    main()

File: c++/web_tutorial_mastercopy/array1d.cpp
#include "Bridges.h"
#include "Array1D.h"

using namespace bridges;

int main(int argc, char **argv) {
	// create Bridges object
#if TESTING
						// command line args provide credentials and server to test on
	Bridges bridges (atoi(argv[1]), argv[2], argv[3]);
	if (argc > 4)
		bridges.setServer(argv[4]);
#else
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");
#endif
	// title, description
	bridges.setTitle("One-Dimensional Array Example");
    bridges.setDescription("OneD array with colors in order left to right: "
			"red, green, blue, cyan, magenta, yellow, red, green, blue, black.");

	// create, populate the array
	Array1D<int> arr (10);

	for (int j = 0; j < 10; j++) {
		arr.getElement(j).setValue(j * j);
		arr.getElement(j).setLabel(to_string(arr.getElement(j).getValue()));
	}

	// set visual attributes
	arr[0].setColor("red");
	arr[1].setColor("green");
	arr[2].setColor("blue");
	arr[3].setColor("cyan");
	arr[4].setColor("magenta");
	arr[5].setColor("yellow");
	arr[6].setColor("red");
	arr[7].setColor("green");
	arr[8].setColor("blue");
	arr[9].setColor("black");

	// visualize
	bridges.setDataStructure(&arr);
	bridges.visualize();

	return 0;
}
